components/Chips/src/MDCChipField.m

```diff
@@ -0,0 +1,573 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCChipField.h"
+
+#import "MaterialTextFields.h"
+
+NSString * const kMDCEmptyTextString = @"";
+NSString * const kMDCChipDelimiterSpace = @" ";
+
+static const CGFloat MDCChipFieldHorizontalInset = 15.f;
+static const CGFloat MDCChipFieldVerticalInset = 8.f;
+static const CGFloat MDCChipFieldIndent = 4.f;
+static const CGFloat MDCChipFieldHorizontalMargin = 4.f;
+static const CGFloat MDCChipFieldVerticalMargin = 5.f;
+static const UIKeyboardType MDCChipFieldDefaultKeyboardType = UIKeyboardTypeEmailAddress;
+
+const CGFloat MDCChipFieldDefaultMinTextFieldWidth = 60.f;
+const UIEdgeInsets MDCChipFieldDefaultContentEdgeInsets = {
+    MDCChipFieldVerticalInset, MDCChipFieldHorizontalInset, MDCChipFieldVerticalInset,
+    MDCChipFieldHorizontalInset};
+
+@protocol MDCChipTextFieldDelegate <NSObject>
+
+- (void)textFieldShouldRespondToDeleteBackward:(UITextField *)textField;
+
+@end
+
+@interface MDCChipTextField : MDCTextField
+
+@property(nonatomic, weak) id<MDCChipTextFieldDelegate> deletionDelegate;
+
+@end
+
+@implementation MDCChipTextField
+
+#pragma mark UIKeyInput
+
+- (void)deleteBackward {
+  [super deleteBackward];
+  if (self.text.length == 0) {
+    [self.deletionDelegate textFieldShouldRespondToDeleteBackward:self];
+  }
+}
+
+#if MDC_CHIPFIELD_PRIVATE_API_BUG_FIX && !(defined(__IPHONE_8_3) && (__IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_8_3))
+
+// WARNING: This is a private method, see the warning in MDCChipField.h.
+// This is only compiled if you explicitly defined MDC_CHIPFIELD_PRIVATE_API_BUG_FIX yourself, and
+// you are targeting an iOS version less than 8.3.
+- (BOOL)keyboardInputShouldDelete:(UITextField *)textField {
+  BOOL shouldDelete = YES;
+  if ([UITextField instancesRespondToSelector:_cmd]) {
+    BOOL (*keyboardInputShouldDelete)(id, SEL, UITextField *) =
+        (BOOL (*)(id, SEL, UITextField *))[UITextField instanceMethodForSelector:_cmd];
+    if (keyboardInputShouldDelete) {
+      shouldDelete = keyboardInputShouldDelete(self, _cmd, textField);
+      BOOL isIos8 = ([[[UIDevice currentDevice] systemVersion] intValue] == 8);
+      BOOL isLessThanIos8_3 = ([[[UIDevice currentDevice] systemVersion] floatValue] < 8.3f);
```

Comment:
Since `systemVersion` is a string and could be "8.3.3", please consider using [`NSProcessInfo`](https://developer.apple.com/documentation/foundation/nsprocessinfo/1414876-isoperatingsystematleastversion?language=objc)

```objectivec
NSOperatingSystemVersion minimumVersion = {8, 3, 0};
NSProcessInfo *processInfo = [NSProcessInfo processInfo];
if ([processInfo isOperatingSystemAtLeastVersion:minimumVersion]) { 
```

Reply:
Fixed.